### DISCUSSION TITLE
Query on Converting SWC to .ANE and Publishing on the App Store

### DISCUSSION THREAD

**Discussion by asuna48** [airsdk/Adobe-Runtime-Support#3598] [Q&A]
Hello Friends,

I hope this message finds you well.

I am reaching out regarding a question for which I could not find clear guidance on "https://airsdk.dev/docs".

Currently, I have successfully converted my iOS mobile application from an SWF file into an IPA format using the "adt" tool and signing it with my Apple Developer Certificate.

Now, I aim to publish this application on the App Store. During my research, I came across some articles suggesting that this could also be achieved using an .ane (Adobe Native Extension).

My question to you is as follows:

Is it possible to directly convert an SWC file into an .ane format? My intention is to utilize this output within Xcode, integrate it with my mobile project, and subsequently publish the application on the App Store.

If you have any documentation, instructions, or guidance on this matter, I would greatly appreciate it. I have found it challenging to locate comprehensive resources on this topic.

Thank you in advance for your assistance and support.

Best regards.

**Comment by marchbold** [comment]
I think you may have interpreted the ANE usage backwards. An ANE is used to wrap code/libraries created with Xcode and make it accessible to an AIR application, not the other way around, i.e. it doesn't wrap an AIR application/libraries to make it accessible in Xcode.

If you could elaborate on your desired outcome perhaps we could suggest an approach?

**Reply by asuna48** [reply]
First of all, thank you for your response and information.

Essentially, I aim to integrate my SWF file into the Xcode environment to develop and publish an application using this platform.

For instance, I am aware that ADT generates IPA files. However, I need access to the source of this IPA or a library that I can utilize within the Xcode environment.

This approach would make it easier to create a mobile application project in Xcode, streamline the process of publishing to the App Store, and enable me to continue running my SWF file.

**Reply by marchbold** [reply]
I'm not aware of a way to run a SWF inside an Xcode application currently, though it has been talked about as an export option. 

Currently though AIR provides an export directly to IPA which you can run in debug mode (on a device or in the ios simulator) and step through your AS3 code as required. It is a great development environment and you simply use transporter to upload your ipa to the app store or you can automate this with scripts.

**Reply by asuna48** [reply]
I understand, thank you again for your response and interest.
